python-backend/app/api/parties/party/resources/party_resource.py

```diff
@@ -72,77 +75,60 @@ def get(self, party_guid=None):
                 'total': pagination_details.total_results,
             }
 
-    def create_party_context(self, party_type_code, party_name, first_name):
-        party_context = {'party_type_code': party_type_code, 'party_name': party_name}
-        if party_type_code == PARTY_STATUS_CODE['per']:
-            if not first_name:
-                self.raise_error(400, 'Error: Party first name is not provided.')
-            party_exists = Party.find_by_name(first_name, party_name)
-            if party_exists:
-                self.raise_error(
-                    400, 'Error: Party with the name: {} {} already exists'.format(
-                        first_name, party_name))
-            party_context.update({
-                'first_name': first_name,
-            })
-        elif party_type_code == PARTY_STATUS_CODE['org']:
-            party_exists = Party.find_by_party_name(party_name)
-            if party_exists:
-                self.raise_error(
-                    400, 'Error: Party with the party name: {} already exists'.format(party_name))
-        else:
-            self.raise_error(400, 'Error: Party type is not provided.')
-        return party_context
-
     @api.expect(parser)
     @requires_role_mine_create
     def post(self, party_guid=None):
         if party_guid:
             self.raise_error(400, 'Error: Unexpected party id in Url.')
         data = PartyResource.parser.parse_args()
-        party_context = self.create_party_context(data['type'], data['party_name'],
-                                                  data['first_name'])
 
         try:
-            party = Party(
-                party_guid=uuid.uuid4(),
-                phone_no=data['phone_no'],
-                email=data['email'],
-                phone_ext=data.get('phone_ext'),
-                **self.get_create_update_dict(),
-                **party_context)
+            party = Party.create(data['party_name'],
+                                 data['email'],
+                                 data['phone_no'],
+                                 data['type'],
+                                 self.get_create_update_dict(),
+                                 # Nullable fields
+                                 first_name=data.get('first_name'),
+                                 phone_ext=data.get('phone_ext'),
+                                 suite_no=data.get('suite_no'),
+                                 address_line_1=data.get('address_line_1'),
+                                 address_line_2=data.get('address_line_2'),
+                                 city=data.get('city'),
+                                 region_code=data.get('region_code'),
+                                 post_code=data.get('post_code'))
         except AssertionError as e:
-            self.raise_error(400, 'Error: {}'.format(e))
```

Comment:
I don't think this will run the '@validates' methods... as the create method adds it to the transaction but doesn't commit the transaction. Let me know if i'm mistaken.  I think this try/except needs to be on the save below, i may be mistaken though.